drivers/counter/timer_tmr_cmsdk_apb.c

```diff
@@ -87,36 +87,60 @@ static u32_t timer_tmr_cmsdk_apb_read(struct device *dev)
 	return value;
 }
 
-static int timer_tmr_cmsdk_apb_set_alarm(struct device *dev,
-					 counter_callback_t callback,
-					 u32_t count, void *user_data)
+static int timer_tmr_cmsdk_apb_set_top_value(struct device *dev,
+					      u32_t ticks,
+					      counter_top_callback_t callback,
+					      void *user_data)
 {
 	const struct timer_tmr_cmsdk_apb_cfg * const cfg =
 						dev->config->config_info;
 	struct timer_tmr_cmsdk_apb_dev_data *data = dev->driver_data;
 
-	if (data->status == TIMER_DISABLED) {
-		return -ENOTSUP;
-	}
-
 	/* Set callback */
 	user_cb = callback;
 
 	/* Store the reload value */
-	data->load = count;
+	data->load = ticks;
 
 	/* Set value register to count */
-	cfg->timer->value = count;
+	cfg->timer->value = ticks;
 
 	/* Set the timer reload to count */
-	cfg->timer->reload = count;
+	cfg->timer->reload = ticks;
 
 	/* Enable IRQ */
 	cfg->timer->ctrl |= TIMER_CTRL_IRQ_EN;
 
 	return 0;
 }
 
+static u32_t timer_tmr_cmsdk_apb_get_top_value(struct device *dev)
+{
+	struct timer_tmr_cmsdk_apb_dev_data *data = dev->driver_data;
+
+	u32_t ticks = data->load;
+
+	return ticks;
+}
+
+static int timer_tmr_cmsdk_apb_set_alarm(struct device *dev, u8_t chan_id,
```

Comment:
same as in other implementations. Function is not needed once `struct counter_config_info` is provided.